modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/GridDistributedTxRemoteAdapter.java

```diff
@@ -800,13 +801,28 @@ else if (op == READ) {
 
                         cctx.mvccCaching().onTxFinished(this, true);
 
+                        boolean walEnabled = false;
+
+                        // Log only there are at least one persistent or cdc enabled group.
                         if (!near() && !F.isEmpty(dataEntries) && cctx.wal() != null) {
```

Comment:
I think we should check `cdcWal()` here too?

Reply:
Fixed. Thanks.